ee/clickhouse/queries/paths/path_event_query.py

```diff
@@ -116,3 +117,29 @@ def _get_event_query(self) -> Tuple[str, Dict[str, Any]]:
             return f" AND {' AND '.join(conditions)}", params
 
         return "", {}
+
+    def _should_query_url(self) -> bool:
```

Comment:
Suggestion: Instead of 2 methods, how about returning an Enum value for representing the 3 cases?

Then based on the enum you can have the cases laid out

Reply:
Enum didn't make much sense to me, since all 3 are independent. There are 2^3 = 8 possible cases (all 3 can exist, none can exist, and any other permutation of the 3)

Reply:
Makes sense. I saw very similar if-else conditions and my mind jumped there.